Source/Artifacts/Artifacts.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Subjects;
+using Microsoft.Extensions.Logging;
+
+namespace Dolittle.SDK.Artifacts
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IArtifacts" />.
+    /// </summary>
+    public class Artifacts : IArtifacts, IDisposable
+    {
+        readonly ReplaySubject<ArtifactAssociation> _registered;
+        readonly BehaviorSubject<IDictionary<Type, Artifact>> _associations;
```

Comment:
Seems a bit superflous to keep track of all the registered artifacts twice? Escpecially if none of them are exposed to the outside world.

We could change the _registered to just be a normal subject, and the _associations to be a BehaviourSubject (which we could allow to outside code to subscribe to for updates.

Reply:
That we can do. Makes sense